src/components/ToDos.jsx

```diff
@@ -0,0 +1,145 @@
+import React, { Component } from 'react';
+import ToDosFooter from './toDos__footer.jsx';
+import ToDosList from './toDos__list.jsx';
+let randomstring = require('randomstring');
+
+export default class ToDos extends Component {
+  state = {
+    data: [],
+    actualData: [],
+    inputValue: ''
+  };
+
+  async componentDidMount() {
+    let localData = JSON.parse(window.localStorage.getItem('localData')) || [];
+
+    this.setState(prevState => ({
+      data: [...prevState.data, ...localData],
+      actualData: [...prevState.data, ...localData]
+    }));
+  }
+
+  addTask = event => {
+    event.preventDefault();
+
+    if (/^ *$/.test(this.state.inputValue)) {
+      return;
+    } else {
+      this.setState(prevState => {
+        let updateData = [
+          ...prevState.actualData,
+          {
+            task: prevState.inputValue,
+            isCompleted: false,
+            id: randomstring.generate(5)
+          }
+        ];
+
+        return {
+          data: updateData,
+          actualData: updateData,
+          inputValue: ''
+        };
+      });
+    }
+  };
+
+  removeTask = id => {
+    this.setState(prevState => {
+      let index = prevState.actualData.findIndex(task => task.id === id);
+      prevState.actualData.splice(index, 1);
```

Comment:
для чого це ? Де зберігається результат виконаної функції?

Reply:
вырезаем из prevState ненужный елемент в 50 и потом присваиваем в новый стэйт, 51 строка

```suggestion
      return {
        data: [...prevState.actualData]
      };
```